.github/ISSUE_TEMPLATE/07-refactor.yml

```diff
@@ -0,0 +1,63 @@
+name: Refactor (Maintainers)
+description: Used to track refactoring opportunities
+title: "Refactor: "
+labels: ["refactor"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Don't forget to [check for existing refactor issue tickets](https://github.com/ggerganov/llama.cpp/issues?q=is%3Aopen+is%3Aissue+label%3Arefactoring) in case it's already covered.
+        Also you may want to check [Pull request refactor label as well](https://github.com/ggerganov/llama.cpp/pulls?q=is%3Aopen+is%3Apr+label%3Arefactoring) for duplicates too.
+
+  - type: textarea
+    id: background-description
+    attributes:
+      label: Background Description
+      description: Please provide a detailed written description of the pain points you are trying to solve.
+      placeholder: Detailed description behind your motivation to request refactor
+    validations:
+      required: true
+
+  - type: dropdown
+    id: grokkability
+    attributes:
+      label: Whats the current grokkability of the code in question
+      multiple: false
+      options:
+        - Easy
+        - Normal
+        - Hard
+        - "=.0"
```

Comment:
These kind of estimates are very subjective - I don't see the value. Something can be easy for somebody and hard for someone else and vice versa